File: web/lib/models/bioskopina.dart
import 'package:json_annotation/json_annotation.dart';
import 'genre_bioskopina.dart';

part 'bioskopina.g.dart';

@JsonSerializable()
class Bioskopina {
  final int id;
  @JsonKey(name: 'titleEn')
  final String titleEn;
  @JsonKey(name: 'titleYugo')
  final String titleYugo;
  final String synopsis;
  final String director;
  @JsonKey(name: 'score')
  final double score;
  @JsonKey(name: 'genreMovies')
  final List<GenreBioskopina> genreMovies;
  @JsonKey(name: 'runtime')
  final int runtime;
  final int yearRelease;
  final String cast;
  @JsonKey(name: 'imDbRatings')
  final String imDbRatings;
  final String? awards;
  @JsonKey(name: 'imageUrl')
  final String? imageUrl;
  @JsonKey(name: 'trailerUrl')
  final String? trailerUrl;

  Bioskopina({
    required this.id,
    required this.titleEn,
    required this.titleYugo,
    required this.synopsis,
    required this.director,
    required this.score,
    required this.genreMovies,
    required this.runtime,
    required this.yearRelease,
    required this.cast,
    required this.imDbRatings,
    this.awards,
    this.imageUrl,
    this.trailerUrl,
  });

  factory Bioskopina.fromJson(Map<String, dynamic> json) =>
      _$BioskopinaFromJson(json);

  Map<String, dynamic> toJson() => _$BioskopinaToJson(this);
}

File: backend/Bioskopina.Services/Database/Bioskopina.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class Bioskopina
{
    public int Id { get; set; }

    public string TitleEn { get; set; } = null!;

   

    public string Synopsis { get; set; } = null!;

    public string? TrailerUrl { get; set; }

    public string? ImageUrl { get; set; }

    public string Director { get; set; } = null!;

    public int YearRelease { get; set; }

    public int Runtime { get; set; }  



    public virtual ICollection<BioskopinaList> BioskopinaLists { get; set; } = new List<BioskopinaList>();

    public virtual ICollection<BioskopinaWatchlist> BioskopinaWatchlists { get; set; } = new List<BioskopinaWatchlist>();

    public virtual ICollection<GenreBioskopina> GenreMovies{ get; set; } = new List<GenreBioskopina>();

    public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();

    public virtual ICollection<Recommender> Recommenders { get; set; } = new List<Recommender>();
    public double Score { get; set; }
}
